.github/workflows/deploy.yml

```diff
@@ -71,12 +82,46 @@ jobs:
         # pytest -m "not accessibility" runs every test which is not marked 
         # accessibility.
         run: python -m pip install pytest && python -m pytest -m "not accessibility" .
+        
+  testing-unit-postgres: 
+    runs-on: ubuntu-latest
+    environment: Dev
+    if: ${{ inputs.postgres_unit_testing == true }}
+    services:
```

Comment:
Is it possible to just create the service conditionally based on the flag so we don't repeat the python stuff between the steps?

Reply:
Whole jobs can be parameterised using matrices but from reading the documentation it seems that services cannot be made conditional.